share/functions/__fish_prepend.fish

```diff
@@ -0,0 +1,68 @@
+function __fish_prepend --description "Prepend/remove 'VALUE ' to/from the beginning of the current commandline"
+    set hist 0  # add previous history item if commandline void
+    set str ""  # actual string to use
+
+    for a in $argv  # arguments reading
+        switch $a
+            case -h --help
+                echo "__fish_prepend - prepend '\$argv[1] ' to the beginning of the current commandline
+usage:
+ __fish_prepend [ARG] VALUE...
+arguments:
+ ARG can be:
+  -h, --help: print this help
+  -p, --previous: add last item from history if commandline is void
+ VALUE: needed (not with -h and --help flag)
+  VALUE to insert at the beginning of the line
+  if the beginning of the line is VALUE, the deletes it
+  otherwise add VALUE at the beginning
+  NOTE: VALUE can be containing spaces
+  NOTE: if gets several VALUEs then concatanates them"
+            case "-p" "--previous"
+                set hist 1
+            case '*'
+                if not test "$str"  # we don't want $str to start with a space
+                    set str $argv[1]
+                else
+                    set -a str $argv[1]
+                end
+        end
+    end
+    
+    if test "$str"  # $str needs to store something
+        if test $hist = 1   # use history ?
+            and not commandline | string length -q
+            commandline -r "$history[1]"
+        end
+
+        set -l cmd (commandline -o)
+        set -l cursor (commandline -C)
+
+        echo "$cmd[1], $str" >> /tmp/fish.elio.output
+        commandline -f repaint
+        # if $cmd[1] is 'sudo' and $str isn't sudo, look at cmd[2]
+        if test "$str" != "sudo"
+            and test "$cmd[1]" = "sudo"
+
+            if echo "$cmd" | sed "s/^$cmd[1] *//g" | grep -q "^$str"
+                commandline -r -- (commandline | sed -z "s/^\(sudo  *\)$str */\1/g")
```

Comment:
Either way I'm not *entirely* sure what you want to do here, I would prefer to do it without `sed`.

The original just does `string sub`, this could also do `string replace`. There's some trickery with multiline arguments, which means you need `string collect` in the correct place.

Reply:
I used `-z` here to avoid sed changing lines that aren't at the very beginning of the buffer (the function as it is now works on the buffer, not only on the current line). I really started scripting in fish not so long ago and I don't know yet every fish functions, that's why I used sed everywhere (I don't know how `string` subcommands work a.e.)

This pr is an occasion for me to learn even more how fish scripting and fish functions work.